.github/workflows/androidBump.yml

```
@@ -0,0 +1,41 @@
+name: Android Rollout Bumper
+
+on:
+  schedule:
+    # Runs at midnight every day
+    - cron: '0 0 * * *'
+
+jobs:
+  android_bump:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node
+        uses: ./.github/actions/composite/setupNode
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1.190.0
+        with:
+          bundler-cache: true
+
+      - name: Decrypt json Google Play credentials
+        run: gpg --batch --yes --decrypt --passphrase="${{ secrets.LARGE_SECRET_PASSPHRASE }}" --output android-fastlane-json-key.json android-fastlane-json-key.json.gpg
+        working-directory: android/app
+
+      - name: Generate version
+        id: checkAndroidStatus
+        uses: ./.github/actions/javascript/checkAndroidStatus
+        with:
+          GOOGLE_KEY_FILE: android/app/android-fastlane-json-key.json
+          PACKAGE_NAME: org.me.mobiexpensifyg
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Update Rollout Percentage with Fastlane
+        # We should not rollout if the release is halted or the rollout percentage is completed
+        if: ${{ steps.checkAndroidStatus.outputs.HALTED != 'true' || steps.checkAndroidStatus.outputs.ROLLOUT_PERCENTAGE != '-1' }}
```

Shouldn't this be an `&&`? We want it to be both not halted and a valid percentage

---

We do not want to bump when it's halted OR an invalid percentage (e.g. it could be halted at day 3 and we don't want to bump), I think that's what the code is doing, right?

---

>We do not want to bump when it's halted OR an invalid percentage

Right, but the code checks for a **valid** percentage `!= -1`. So right now, if we have `halted=true`, the first condition would return false (that's correct, we don't want to bump when halted), but if `ROLLOUT_PERCENTAGE = 0.01`, the 2nd condition would return true , the whole thing would evaluate to `true` and we'd end up bumping the version, when we shouldn't.

I think we should change this to:

```suggestion
        if: ${{ steps.checkAndroidStatus.outputs.HALTED == 'false' && steps.checkAndroidStatus.outputs.ROLLOUT_PERCENTAGE != '-1' }}
```